ls-ui-react-app/src/pages/UploadCircuitPage.tsx

```diff
@@ -207,10 +222,51 @@ const UploadCircuitPage = ( {appState, setAppState} : AppStateProps)  =>
         <div className='main'>
             <section>
                 <div>
+                    {/* Indicate loading/calculating with a spinner and text */}
                     <CompilerInputCircuitSelection appState={appState} setAppState={setAppState} />
-                    { appState.compilationIsLoading && <b> Loading... </b> }
+                    { appState.compilationIsLoading && 
+                        <div>
+                            <div className="spinner-border text-success" role="status"></div>
+                            <b>Processing...</b>
+                        </div>
+                    }
+
+                    {/* If a request has been made, and compilation has finished, and compilation is still undefined, indicate error message */}
+                    { (appState.request === true && appState.compilationIsLoading === false && appState.compilationResult === undefined) && 
+                        <div>
+                            
+                            {/* AXIOS error, ie, no response*/}
+                            {appState.errorMsg && <div className="alert alert-danger">
+                                <div>{appState.errorMsg}</div>
+                            </div>}
+                            
+                            {/* RESPONSE returned, but non-200 */}
+                            { (appState.apiResponse.status  && appState.apiResponse.status !=200) &&
+                                <div className="alert alert-danger" role="alert">
+                                    <div>API Status: {appState.apiResponse.status}</div>
+                                </div>
+                            }
+
+                            {/* Response 200, but Compiler Error */}
+                            { appState.apiResponse.data && 
+                                <div className="alert alert-danger">
+                                    {appState.apiResponse.data.errorType + ": " + appState.apiResponse.data.errorMessage}
+                                </div>
+                            }
+
```

Comment:
This is some very clear handling of error cases I can see everything that's happening